lib/rubocop/ast/node/self_node.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module AST
+    # A node extension for `self` nodes. This will be used in place of a
+    # plain node when the builder constructs the AST, making its methods
+    # available to all `self` nodes within RuboCop.
+    class SelfNode < Node
+      include MethodIdentifierPredicates
+      include ParameterizedNode
```

Comment:
I don't think `self` can accept arguments. 🙂 

```
ruby-parse -e 'self(1)'
#=> error: unexpected token tLPAREN2
```

Reply:
Sure! I updated the PR.